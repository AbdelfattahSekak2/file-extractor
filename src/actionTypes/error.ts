export const SetError = {
  DEFAULT: "SET_ERROR_DEFAULT",
  CLEAR: "SET_ERROR_CLEAR",
}
