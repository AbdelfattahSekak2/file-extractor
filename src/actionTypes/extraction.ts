export const GetExtraction = {
  FAILURE: "GET_EXTRACTION_FAILURE",
  SUCCESS: "GET_EXTRACTION_SUCCESS",
  REQUEST: "GET_EXTRACTION_REQUEST",
}

export const ClearExtractionState = {
  DEFAULT: "CLEAR_EXTRACTION_STATE_DEFAULT",
}
