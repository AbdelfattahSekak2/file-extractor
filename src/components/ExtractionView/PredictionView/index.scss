.prediction-container {
  margin-right: 10px;
  display: flex;
  flex-direction: column;
  width: 100%;
  .title {
    color: #aaa7a7;
    font-weight: 700;
    font-size: 22px;
    margin: 10px;
    text-align: center;
  }
}
