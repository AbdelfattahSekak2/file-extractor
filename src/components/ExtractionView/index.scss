@import "../../styles/mixins";

.extraction-container {
  display: flex;
  flex-direction: column;
  position: relative;
  padding: 10px;
  width: 30%;
  margin: auto;
  h1 {
    font-size: 30px;
    font-weight: 500;
    color: $dark-grey;
    text-align: center;
    margin: 0;
  }
  .ui.dimmer {
    position: fixed;
  }
  .slick-slider {
    .slick-list {
      width: 100%;
    }
    .slick-arrow {
      &::before {
        font-size: 40px;
      }
      &.slick-prev {
        left: -50px;
      }
    }
  }
}
