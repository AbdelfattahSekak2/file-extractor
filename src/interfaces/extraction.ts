export interface Coordinates {
  coord_x1: number;
  coord_x2: number;
  coord_y1: number;
  coord_y2: number;
}

export interface Tax extends Coordinates {
  rate: number;
  amount: number;
  probability: number;
}

interface Date extends Coordinates {
  date_INTL: string,
  date_US: string,
  probability: 0.972,
}

interface Total extends Coordinates {
  amount: number,
  probability: number,
}

export interface Prediction {
  orientation: {
    degrees: string;
    probability: number;
  },
  category: {
    value: string;
    probability: number;
  },
  date: Date
  total: Total;
  taxes: Tax[]
}

export default interface Extraction {
  details: {
    call_id: string;
    computation_time: number;
  };
  predictions: Prediction[];
}
