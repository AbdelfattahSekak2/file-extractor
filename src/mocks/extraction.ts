import Extraction from "../interfaces/extractions";

export const extraction: Extraction = {
  "details": {
    "call_id": "49ef6b42-2724-4eaf-80bf-9add06a45176",
    "computation_time": 0.789,
  },
  "predictions": [
    {
      "orientation": {
        "degrees": "0",
        "probability": 0.723
      },
      "category": {
        "value": "parking",
        "probability": 0.302
      },
      "date": {
        "date_INTL": "2018-01-02",
        "date_US": "2012-10-26",
        "probability": 0.972,
        "coord_x1": 0.865,
        "coord_x2": 0.18,
        "coord_y1": 0.249,
        "coord_y2": 0.818
      },
      "total": {
        "amount": 0.732,
        "probability": 0.801,
        "coord_x1": 0.255,
        "coord_x2": 0.2,
        "coord_y1": 0.461,
        "coord_y2": 0.927,
      },
      "taxes": [
        {
          "rate": 0.476,
          "amount": 0.57,
          "probability": 0.647,
          "coord_x1": 0.216,
          "coord_x2": 0.256,
          "coord_y1": 0.16,
          "coord_y2": 0.145,
        }
      ]
    },
  ]
}
