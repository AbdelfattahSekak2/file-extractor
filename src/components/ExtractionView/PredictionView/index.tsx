import * as React from "react";

import { Prediction, Tax } from "../../../interfaces/extraction";

import "./index.scss";
import PredictionItem from "./PredictionItem";

interface IProps {
  prediction: Prediction;
}

const PredictionView: React.FunctionComponent<IProps> = ({
  prediction,
}) => {
  const { orientation, taxes, date, total, category } = prediction;
  return (
    <div className="prediction-container" >
      {
        orientation &&
        <PredictionItem
          label="Orientation"
          element={{
            probability: orientation.probability,
            sign: "°",
            value: orientation.degrees,
          }}
        />
      }
      {
        category &&
        <PredictionItem
          label="Category"
          element={{
            probability: category.probability,
            value: category.value,
          }}
        />
      }
      {
        date &&
        <PredictionItem
          label="Date"
          element={{
            probability: date.probability,
            value: date.date_INTL,
          }}
        />
      }
      {
        total &&
        <PredictionItem
          label="Total"
          element={{
            probability: total.probability,
            sign: "€",
            value: total.amount,
          }}
        />
      }
      {
        taxes && taxes.length > 0 &&
        <PredictionItem
          label="Taxes"
          element={taxes.map((e: Tax) => ({
            probability: e.probability,
            sign: "€",
            subValue: {
              label: "Rate",
              value: e.rate,
            },
            value: e.amount,
          }))}
        />
      }
    </div >
  );
};

export default PredictionView;
