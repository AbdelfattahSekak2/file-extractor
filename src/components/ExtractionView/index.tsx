import * as React from "react";
import Slider from "react-slick";
import { Dimmer, Loader } from "semantic-ui-react";
import "slick-carousel/slick/slick-theme.css";
import "slick-carousel/slick/slick.css";

import Extraction, { Prediction } from "../../interfaces/extraction";

import { GetExtraction } from "../../actionTypes/extraction";
import { ApiState } from "../../interfaces/apiState";
import "./index.scss";
import PredictionView from "./PredictionView";

interface IProps {
  apiPendingRequests: ApiState[];
  extraction: Extraction | null;
}

const settings = {
  dots: true,
  infinite: true,
  slidesToScroll: 1,
  slidesToShow: 1,
  speed: 500,
};

const ExtractionView: React.FunctionComponent<IProps> = ({
  extraction,
  apiPendingRequests,
}) => {
  let content: React.ReactElement | null = null;
  const loading = apiPendingRequests.some((item: ApiState) => item.type === GetExtraction.REQUEST);
  if (loading) {
    content = <Dimmer active>
      <Loader size="huge" indeterminate>Preparing your file</Loader>
    </Dimmer>;
  } else if (extraction && extraction.predictions && extraction.predictions.length > 0) {
    content = <div className="extraction-container">
      <h1>Predictions</h1>
      <Slider {...settings}>
        {
          extraction.predictions.map(
            (item: Prediction, key: number) => <PredictionView key={key} prediction={item} />,
          )
        }
      </Slider>
    </div>;
  }
  return content;
};

export default ExtractionView;
