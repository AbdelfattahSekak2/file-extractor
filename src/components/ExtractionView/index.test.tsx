import React from "react";

import ExtractionView from "."
import { ShallowWrapper, shallow } from "enzyme";
import { extraction } from "../../mocks/extraction";
import { GetExtraction } from "../../actionTypes/extraction";

describe("ExtractionView component", () => {
  const defaultProps = {
    apiPendingRequests: [],
    extraction,
  }
  let wrapper: ShallowWrapper;
  it("should render correctly", () => {
    wrapper = shallow(<ExtractionView {...defaultProps} />);
    expect(wrapper).not.toBeNull();
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot when it's data loading", () => {
    const apiPendingRequests = [{ type: GetExtraction.REQUEST }];
    wrapper = shallow(<ExtractionView {...defaultProps} apiPendingRequests={apiPendingRequests} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot when extraction is null", () => {
    wrapper = shallow(<ExtractionView apiPendingRequests={[]} extraction={null} />);
    expect(wrapper).toMatchSnapshot();
  });
})
