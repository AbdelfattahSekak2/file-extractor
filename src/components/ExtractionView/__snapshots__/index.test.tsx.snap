// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtractionView component should match snapshot when extraction is null 1`] = `""`;

exports[`ExtractionView component should match snapshot when it's data loading 1`] = `
<Dimmer
  active={true}
>
  <Loader
    indeterminate={true}
    size="huge"
  >
    Preparing your file
  </Loader>
</Dimmer>
`;

exports[`ExtractionView component should render correctly 1`] = `
<div
  className="extraction-container"
>
  <h1>
    Predictions
  </h1>
  <Slider
    dots={true}
    infinite={true}
    slidesToScroll={1}
    slidesToShow={1}
    speed={500}
  >
    <PredictionView
      key="0"
      prediction={
        Object {
          "category": Object {
            "probability": 0.302,
            "value": "parking",
          },
          "date": Object {
            "coord_x1": 0.865,
            "coord_x2": 0.18,
            "coord_y1": 0.249,
            "coord_y2": 0.818,
            "date_INTL": "2018-01-02",
            "date_US": "2012-10-26",
            "probability": 0.972,
          },
          "orientation": Object {
            "degrees": "0",
            "probability": 0.723,
          },
          "taxes": Array [
            Object {
              "amount": 0.57,
              "coord_x1": 0.216,
              "coord_x2": 0.256,
              "coord_y1": 0.16,
              "coord_y2": 0.145,
              "probability": 0.647,
              "rate": 0.476,
            },
          ],
          "total": Object {
            "amount": 0.732,
            "coord_x1": 0.255,
            "coord_x2": 0.2,
            "coord_y1": 0.461,
            "coord_y2": 0.927,
            "probability": 0.801,
          },
        }
      }
    />
  </Slider>
</div>
`;
